File: AL/nadongbin/01-greedy/untilValueReachesOne/untilValueReachesOne.ts
function untilValueReachesOne(N: number, K: number) {
  let count = 0

  while (true) {
    let target = Math.floor(N / K) * K
    count += N - target
    N = target

    if (N < K) {
      break
    }

    count++
    N = Math.floor(N / K)
  }

  count += N - 1
  return count
}

console.log(untilValueReachesOne(15, 5))

File: AL/nadongbin/01-greedy/untilValueReachesOne/untilValueReachesOne.go
package main

import "fmt"

func main() {
	N, K := 17, 3
	fmt.Println(untilValueReachesOne(N, K))
}

func untilValueReachesOne(N int, K int) int {
	count := 0
	for N >= K {
		count += N % K
		N = N / K
		count++
	}

	count += N - 1

	return count
}
